src/fabric.c

```diff
@@ -254,11 +255,13 @@ static void ofi_filter_info(struct fi_info **info)
 	}
 }
 
-static struct ofi_prov *ofi_getprov(const char *prov_name, size_t len)
+static struct ofi_prov *ofi_getprov(const char *prov_name, size_t len,
+                                    bool hidden)
 {
 	struct ofi_prov *prov;
 
-	for (prov = prov_head; prov; prov = prov->next) {
+	for (prov = hidden ? prov_head_hidden : prov_head;
```

Comment:
I assumed you were going to have a separate list of providers that qualified to be hidden when excluded, but I like this even better.